### DISCUSSION TITLE
useLongPress interaction for touch devices?

### DISCUSSION THREAD

**Discussion by aeharding** [floating-ui/floating-ui#2368] [Q&A] [answered]
Hi there! Great library.

"Long press" as a trigger for popovers is a common pattern in iOS (see: calendar app, photos app, etc).

Is there a hook available to accomplish this? Bonus points for falling back to contextmenu (right click) on pointer/hover (i.e. desktop) devices.

Thanks very much!
Alex

**Comment by atomiks** [comment, marked as answer]
There's not, but was being tracked here: https://github.com/floating-ui/floating-ui/issues/1728

You can make a custom hook to achieve it, if you make one feel free to make a PR 😊

**Reply by aeharding** [reply]
Thanks for that! I was actually trying to create a custom hook earlier, but I was struggling to create one, I just couldn't get the logic working nicely without clobbering other events.
